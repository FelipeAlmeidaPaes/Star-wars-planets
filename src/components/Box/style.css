.box {
  background: #f3f3f3;
  border-radius: 10px;
  width: 300px;
  height: fit-content;
}

.planet_name {
  padding: 20px 0 0 0;
  margin: 0;
  text-align: center;
}

hr {
  margin: 20px 0;
  padding: 0; 
}

.planet_population,
.planet_climate,
.planet_terrain {
  padding: 0 20px 10px 20px;
}

.planet_films {
  text-align: center;
  padding-bottom: 20px;
}

.button {
  background: #fff;
  border: 0 solid #fff;
  border-radius: 10px;
  padding: 20px;
  width: 50%;
  margin-left: 50%;
  transform: translateX(-50%);
  outline: none;
  transition: .3s;
  font-size: 14px;
  font-weight: bold;
}

.button:hover {
  background: #656565;
  color: #fff;
}