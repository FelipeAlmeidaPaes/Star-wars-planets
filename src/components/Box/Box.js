import React, { useEffect, useState } from 'react';
import api from '../../services/api';
import './style.css';

export default function Box() {
  const [loading, setLoading] = useState(true)
  const [planet, setPlanet] = useState({ films: [] })
  const [numberOfPlanets, setNumberOfPlanets] = useState(0)
  let [random, setRandom] = useState(Math.floor(parseInt(Math.random() * (numberOfPlanets) + 1)))

  const getNumberOfPlanets = async () => {
    const allPlanetsResponse = await api.get()
    const count = allPlanetsResponse.data.count

    setNumberOfPlanets(count)
  }
  getNumberOfPlanets()

  useEffect(() => {
    
    setLoading(true)
    async function loadPlanet() {
      const response = await api.get(`${random}`)

      setPlanet(response.data)
      setLoading(false)
    }
    loadPlanet()

  }, [random])

  return (
    <>
      <div className="box">
        <h2 className="planet_name">{!loading ? planet.name : 'Loading...'}</h2>
        <hr />

        <div className="planet_population"><b>Population</b>: {!loading ? planet.population : 'Loading...'}</div>
        <div className="planet_climate"><b>Climate</b>: {!loading ? planet.climate : 'Loading...'}</div>
        <div className="planet_terrain"><b>Terrain</b>: {!loading ? planet.terrain : 'Loading...'}</div>

        <p className="planet_films">Featured in <b>{!loading ? planet.films.length : 'x'}</b> films.</p>
      </div>

      <button className="button" onClick={() => setRandom(Math.floor(parseInt(Math.random() * (numberOfPlanets) + 1)))}>Next</button>
    </>
  );
};
