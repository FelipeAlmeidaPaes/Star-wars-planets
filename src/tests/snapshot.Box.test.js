import React from 'react'
import Box from "../components/Box/Box";
import renderer from "react-test-renderer";

it('renders correctly', () => {
  const tree = renderer
    .create(<Box />).toJSON();
  expect(tree).toMatchSnapshot();

})