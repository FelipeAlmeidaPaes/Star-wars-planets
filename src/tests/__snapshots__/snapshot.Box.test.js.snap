// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="box"
  >
    <h2
      className="planet_name"
    >
      Loading...
    </h2>
    <hr />
    <div
      className="planet_population"
    >
      <b>
        Population
      </b>
      : 
      Loading...
    </div>
    <div
      className="planet_climate"
    >
      <b>
        Climate
      </b>
      : 
      Loading...
    </div>
    <div
      className="planet_terrain"
    >
      <b>
        Terrain
      </b>
      : 
      Loading...
    </div>
    <p
      className="planet_films"
    >
      Featured in 
      <b>
        x
      </b>
       films.
    </p>
  </div>,
  <button
    className="button"
    onClick={[Function]}
  >
    Next
  </button>,
]
`;
